src/instructlab/profile/list.py

```diff
@@ -0,0 +1,44 @@
+# Standard
+from typing import Any, Dict, List
+import os
+
+
+def list_profiles(startpath: str) -> Dict[str, Any]:
```

Comment:
Confused by this.

Isn't it just `[path.split(os.sep)[-1] for path in glob.glob("src/instructlab/profiles/*")]` ?

Reply:
@reidliu41 - can you PTAL?

thanks!

Reply:
yeah, similar, maybe more:
```
  "apple": {
        "m1": {
            "files": [
                "m1_ultra.yaml",
                "m1_max.yaml"
            ]
        },
        "m2": {
            "files": [
                "m2_max.yaml",
                "m2_ultra.yaml",
                "m2_pro.yaml",
                "m2.yaml"
            ]
        },
        "m3": {
            "files": [
                "m3_max.yaml",
                "m3_pro.yaml",
                "m3.yaml"
            ]
        }
    },
    "intel": {
        "files": [
            "cpu.yaml"
        ],
        "gaudi": {
            "files": [
                "gaudi_3.yaml"
            ]
        }
    },
```

Reply:
Ah I see what you mean. Could you please expand the test case to include some more checks for what this command tries to achieve (going deep into the hierarchy)? It's not clear from what the current test case suggests.